kikuchipy/indexing/static_dictionary_indexing.py

```diff
@@ -0,0 +1,179 @@
+            Dictionaries as EBSD Signals with one-dimensional navigation axis
+            and with the `xmap` property set.
+        """
+        self.dictionaries = (
+            dictionaries if isinstance(dictionaries, list) else [dictionaries]
+        )
+
+    def index(
+        self,
+        patterns: Union["EBSD", "LazyEBSD"],
+        metric: Union[str, SimilarityMetric] = "zncc",
+        keep_n: int = 1,
+        n_slices: int = 1,
+        merge_xmaps: bool = True,
+        osm: bool = True,
+    ) -> List[CrystalMap]:
+        """Perform Dictionary Indexing on patterns against preloaded dictionaries.[ref here or elsewhere?]
+
+        Produce a `CrystalMap` for each dictionary with `metric_results` and `template_indices` as properties.
+
+        Parameters
+        ----------
+        patterns : Union[EBSD, LazyEBSD]
+            Patterns
+        metric : Union[str, SimilarityMetric], optional
+            Similarity metric, by default "zncc".
+        keep_n : int, optional
+            Number of sorted results to keep, by default 1.
+        n_slices : int, optional
+            Number of template slices to process sequentially, by default 1.
+        merge_xmaps : bool, optional
+            Produce a merged crystal map from best results, by default True.
+            See also `merge_crystalmaps`.
+        osm : bool, optional
+            Orientation Similarity Maps as property `osm`, by default True.
+
+        Returns
+        -------
+        xmaps : List[CrystalMap]
+            A crystal map for each dictionary loaded and one merged map if `merge_xmaps = True`.
+        """
+
+        # This needs a rework before sent to cluster and possibly more automatic slicing with dask
+        num_templates = self.dictionaries[0].data.shape[0]
+        if num_templates // n_slices > 13500:
+            answer = input(
+                f"You should probably increase n_slices depending on your available memory, try above {num_templates // 13500}. Do you want to proceed? [y/n]"
+            )
+            if answer != "y":
+                return
+
+        n_slices = None if n_slices == 1 else n_slices
+
+        metric = SIMILARITY_METRICS.get(metric, metric)
+
+        # Naively let dask compute them seperately, should try in the future combined compute for better performance
+        match_results = [
+            pattern_match(
+                patterns.data,
+                dictionary.data,
+                metric=metric,
+                keep_n=keep_n,
+                n_slices=n_slices,
+            )
+            for dictionary in self.dictionaries
+        ]
+
+        axm = patterns.axes_manager
+        scan_unit = axm.navigation_axes[0].units
+        x1, x2, y1, y2 = axm.navigation_extent
+        scale_x, scale_y = (axm.navigation_axes[i].scale for i in range(2))
+        nav_shape = axm.navigation_shape
+        x = np.tile(np.arange(x1, x2 + scale_x, scale_x), nav_shape[1])
```

Comment:
Yeah, this should be a convenience function in the `orix.crystal_map` module, to just create spatial arrays compatible with `CrystalMap`. (Resolve when read.)